File: assets/scripts/framework/native/JsBridge.ts
enum JsEventType {
    QiyuSDK = "QiyuSDK"
}

function js_callback(eventName: string, eventData: string): void {
    if (eventName === JsEventType.QiyuSDK) {
        const jsonObj = JSON.parse(eventData);
        console.log(`cocos 收到原生消息: ${jsonObj.count}, ${jsonObj.message}`);
    }
}

window.js_callback = js_callback;

File: native/engine/android/app/src/com/cocos/game/JsBridge.java
package com.cocos.game;

import android.util.Log;

import com.cocos.lib.CocosHelper;
import com.cocos.lib.CocosJavascriptJavaBridge;

import org.json.JSONException;
import org.json.JSONObject;

public class JsBridge {
    private static final String TAG = "JsBridge";
    private static final String JS_CALLBACK_FORMAT = "window.js_callback('%s', '%s')";

    private static void onJsCallback(String eventName, String eventData) {
        eventData = eventData.replace("'", "\\'");
        final String jsCode = String.format(JS_CALLBACK_FORMAT, eventName, eventData);

        CocosHelper.runOnGameThread(() -> CocosJavascriptJavaBridge.evalString(jsCode));
    }

    public static void onQiyuNewMessage(String message) {
        try {
            JSONObject obj = new JSONObject();
            obj.put("count", 1);
            obj.put("message", message);

            onJsCallback(JsEventType.QIYU_SDK, obj.toString());
        } catch (JSONException e) {

        }
    }
}
